### PR TITLE
Update to new interceptor api

### PR DESCRIPTION
**Pull request by ash-lshift**
An example of how the API changes in https://github.com/rabbitmq/rabbitmq-server/pull/214 work.

### CODE REVIEW

File: test/src/rabbit_hash_exchange_test.erl [hunk 1]
```diff
@@ -62,10 +62,14 @@ test0(MakeMethod, MakeMsg, Queues, MsgCount, Count) ->
...
      || Q <- Queues],
 
+    amqp_channel:call(Chan, #'confirm.select'{}),
+
     [amqp_channel:call(Chan,
                        MakeMethod(),
                        MakeMsg()) || _ <- lists:duplicate(MsgCount, const)],
 
+    amqp_channel:wait_for_confirms_or_die(Chan),
+
```
Comment (videlalvaro) [inline test/src/rabbit_hash_exchange_test.erl]: what's the point of these changes on the tests here?

Comment (ash-lshift) [inline test/src/rabbit_hash_exchange_test.erl]: I was getting intermittent failures and I thought maybe there was a race between publishing the message and getting the number of messages on the queue. Does that sound plausible?

Comment (videlalvaro) [inline test/src/rabbit_hash_exchange_test.erl]: Yes it does, only with confirms I think you get the right number, or maybe using transactions which for this case is the same.

Comment (ash-lshift) [inline test/src/rabbit_hash_exchange_test.erl]: I've added a comment to that effect.

**Comment by michaelklishin** [comment]
@ash-lshift @videlalvaro any updates on this?

**Comment by videlalvaro** [comment]
@michaelklishin once the new interceptor API is part of the broker, then we can merge this PR. So we have to wait for 3.6.0

**Comment by michaelklishin** [comment]
Why? Can't we just tag/branch off a version for 3.5.x releases in this repo?

Most users should use binary builds of that branch, not build from master.

**Comment by videlalvaro** [comment]
So you suggest I create a new tag, or a new branch, or both?

**Comment by michaelklishin** [comment]
For `rabbitmq-clusterer` we added a branch (`stable`) only.

**Pull request merged**